libbeat/common/safemapstr/safemapstr.go

```diff
@@ -0,0 +1,66 @@
+package safemapstr
+
+import (
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+// Put This method implements a way to put dotted keys into a MapStr while
+// ensuring they don't override each other. For example:
+//
+//  a := MapStr{}
+//  safemapstr.Put(a, "com.docker.swarm.task", "x")
+//  safemapstr.Put(a, "com.docker.swarm.task.id", 1)
+//  safemapstr.Put(a, "com.docker.swarm.task.name", "foobar")
+//
+// Will result in `{"com":{"docker":{"swarm":{"task":{"id":1,"name":"foobar","value":"x"}}}}}`
+//
+// Put detects this scenario and renames the common base key, by appending
+// `.value`
+func Put(data common.MapStr, key string, value interface{}) error {
```

Comment:
how about introducing a 'defaultKey' to be used instead of hard-coded 'value' ?

Reply:
do you mean moving it to a `const` or passing it as a parameter to `Put`?

Reply:
parameter. I wonder if in some cases we can declare/define a better name then 'value'.